### PR TITLE
transformer/markdown: url-encode href attributes

### PR DESCRIPTION
**Pull request by alxndrsn**

### CODE REVIEW

File: packages/enketo-transformer/test/markdown.spec.ts [hunk 1]
```diff
@@ -68,6 +68,18 @@ describe('markdown', () => {
                 '[link](http://example.org)',
                 '<a href="http://example.org" rel="noopener" target="_blank">link</a>',
             ],
+            [
+                '[link](http://example.org/cgi-bin/Markdown.pl?a=1&b=2)',
+                '<a href="http://example.org/cgi-bin/Markdown.pl?a=1&b=2" rel="noopener" target="_blank">link</a>',
+            ],
+            [
+                '[link with a "quote" inside](http://example.org)',
```
Comment (lognaturel) [inline packages/enketo-transformer/test/markdown.spec.ts:76]: Did this case intend to have a quote inside?

Comment (alxndrsn) [reply packages/enketo-transformer/test/markdown.spec.ts:76]: only in the link text... would `link text with "quotes" inside` be clearer?

Comment (lognaturel) [reply packages/enketo-transformer/test/markdown.spec.ts:76]: The intent is to verify that the label is not uri-encoded?

**Review by lognaturel — APPROVED** [review]
I think there's some possibility that forms exist with URLs that have manually URL-encoded parts and those will now be double-encoded and stop working. That's unfortunate but this should have been the initial behavior so it feels right to change.

**Pull request merged**